pkg/specs/replicatedapp/local.go

```diff
@@ -0,0 +1,117 @@
+// local.go has methods for resolving a local ship.yaml file, and patching in api.Release info
+// that would usually be returned by pg.replicated.com
+package replicatedapp
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/ship/pkg/constants"
+)
+
+func (r *resolver) resolveRunbookRelease() (*ShipRelease, error) {
+	debug := level.Debug(log.With(r.Logger, "method", "resolveRunbookRelease"))
+	debug.Log("phase", "load-specs", "from", "runbook", "file", r.Runbook)
+
+	specYAML, err := r.FS.ReadFile(r.Runbook)
+	if err != nil {
+		return nil, errors.Wrapf(err, "read specs from %s", r.Runbook)
+	}
+	debug.Log("phase", "load-specs", "from", "runbook", "file", r.Runbook, "spec", specYAML)
+
+	if err := r.persistSpec(specYAML); err != nil {
+		return nil, errors.Wrapf(err, "serialize last-used YAML to disk")
+	}
+	debug.Log("phase", "write-yaml", "from", r.Runbook, "write-location", constants.ReleasePath)
+
+	fakeGithubContents, err := r.loadLocalGitHubContents()
```

Comment:
this method moved from `resolver.go` this line is the main addition, everything above it is unchanged